### PR TITLE
[runtime-security] Support approvers on all events

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

It adds support for basename approvers on most events

### Motivation

Right now, approvers were only used on open events. That means we only relied on discarders
for all the other events. If we can find approvers for one event type, it avoids sending all the events
for this event type to userspace.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
Gitlab/invoke changes LGTM

File: pkg/security/ebpf/c/link.h [hunk 1]
```diff
@@ -40,13 +47,18 @@ int kprobe__vfs_link(struct pt_regs *ctx) {
     if (!syscall)
         return 0;
 
+    // if second pass, ex: overlayfs, just cache the inode that will be used in ret
```
Comment (safchain) [inline pkg/security/ebpf/c/link.h:50]: nit: not sure to understand this comment

File: pkg/security/ebpf/c/rmdir.h [hunk 2]
```diff
@@ -94,14 +108,12 @@ SYSCALL_KRETPROBE(rmdir) {
...
     int retval = PT_REGS_RC(ctx);
-
     if (IS_UNHANDLED_ERROR(retval)) {
-        invalidate_inode(ctx, syscall->rmdir.path_key.mount_id, syscall->rmdir.path_key.ino, 0);
         return 0;
     }
 
     int enabled = is_event_enabled(EVENT_RMDIR);
-    if (enabled) {
+    if (!syscall->discarded && enabled) {
```
Comment (safchain) [inline pkg/security/ebpf/c/rmdir.h:116]: should we not use the same test for the following invalidate_inode ?

File: pkg/security/probe/approvers_basename_bpf.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+package probe
+
+import (
+	"path"
+
+	"github.com/DataDog/datadog-agent/pkg/security/rules"
+)
+
+func onNewBasenameApprovers(probe *Probe, eventType EventType, field string, approvers rules.Approvers) ([]activeApprover, error) {
```
Comment (safchain) [inline pkg/security/probe/approvers_basename_bpf.go:16]: nit: I think you can remove the `_bpf` to this file ?

File: pkg/security/ebpf/c/rmdir.h (hunk 2 above)
Comment (lebauce) [reply pkg/security/ebpf/c/rmdir.h:116]: Indeed. Fixed

File: pkg/security/probe/approvers_basename_bpf.go (hunk 3 above)
Comment (lebauce) [reply pkg/security/probe/approvers_basename_bpf.go:16]: Done

**Review by safchain — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
lgtm :+1:

**Pull request merged**